src/lib/transition_handler/validator.ml

```diff
@@ -81,28 +90,56 @@ module Make (Inputs : Inputs.S) = struct
       *)
           let%map proof_is_valid =
             State_proof.verify
-              (External_transition.protocol_state_proof t)
-              (External_transition.protocol_state t)
+              (External_transition.protocol_state_proof transition)
+              (External_transition.protocol_state transition)
           in
           log_assert proof_is_valid "proof was invalid"
       | x -> return x
 
+  let verify_transition ~staged_ledger ~transition :
```

Comment:
this will become `Deffered` after you re-enable the Async.Thread_safe

Reply:
Ok